File: Easy/Power of Thor/Java/PowerOfThor.java
import java.util.*;

class Player {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int lightX = in.nextInt(); // the X position of the light of power
        int lightY = in.nextInt(); // the Y position of the light of power
        int thorX = in.nextInt(); // Thor's starting X position
        int thorY = in.nextInt(); // Thor's starting Y position

        while (true) {

            String directionX = "";
            if (lightX > thorX) { thorX++; directionX = "E"; } else if (lightX < thorX) { thorX--; directionX = "W"; }

            String directionY = "";
            if (lightY > thorY) { thorY++; directionY = "S"; } else if (lightY < thorY) { thorY--; directionY = "N"; }

            System.out.println((directionY) + (directionX));
        }
    }
}

File: Easy/Power of Thor/C++/Power of Thor.cpp
#include <iostream>

using namespace std;

int main()
{
	int lightX;
	int lightY;
	int thorX;
	int thorY;
	cin >> lightX >> lightY >> thorX >> thorY; cin.ignore();

	while (1) {
		string directionX = "";
		string directionY = "";

		if (thorX < lightX)
		{
			directionX = "E";
			thorX++;
		}
		else if (thorX > lightX)
		{
			directionX = "W";
			thorX--;
		}

		if (thorY < lightY)
		{
			directionY = "S";
			thorY++;
		}
		else if (thorY > lightY)
		{
			directionY = "N";
			thorY--;
		}

		cout << directionY << directionX << endl;
	}
}